### PR TITLE
chore: simplify repo configuration for infra repos

### PR DESCRIPTION
**Pull request by garethjevans**
This also enabled TrustNobody on the configured repos, which may have
some unforeseen side effects.

#### Is this a new chart
- [ ] `CODEOWNERS` is updated
- [ ] If secrets are needed, `.sops.yaml` must be updated with appropriated public gpg key ID, and full public gpg key must be provided to @olblak

#### Which issue this PR fixes

  - fixes # [INFRA-XXXX](https://issues.jenkins-ci.org/projects/INFRA/issues/INFRA-XXXX)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: config/default/jenkins-infra.yaml [hunk 1]
```diff
@@ -149,104 +150,48 @@ jenkins:
...
+                      def traits = node / 'sources' / 'data' / 'jenkins.branch.BranchSource' / 'source' / 'traits'
+                      traits << 'org.jenkinsci.plugins.github__branch__source.BranchDiscoveryTrait' {
+                        strategyId(1)
+                      }
+                      traits << 'org.jenkinsci.plugins.github__branch__source.OriginPullRequestDiscoveryTrait' {
+                        strategyId(1)
+                      }
+                      traits << 'org.jenkinsci.plugins.github__branch__source.ForkPullRequestDiscoveryTrait' {
+                        strategyId(1)
+                        trust(class: 'org.jenkinsci.plugins.github_branch_source.ForkPullRequestDiscoveryTrait$TrustNobody')
```
Comment (timja) [inline config/default/jenkins-infra.yaml:186]: why? this makes it so that Jenkinsfile changes can't be tested by anyone from fork PRs afaik

Comment (garethjevans) [reply config/default/jenkins-infra.yaml]: changed to TrustPermission

Comment (garethjevans) [reply config/default/jenkins-infra.yaml]: and made it configurable per repo

File: config/default/jenkins-infra.yaml [hunk 2]
```diff
@@ -149,104 +150,48 @@ jenkins:
...
-                  factory {
-                    workflowBranchProjectFactory {
-                      scriptPath('Jenkinsfile_k8s')
+                [
+                  [repo: 'docker-packer', owner: 'jenkins-infra', name: 'Packer Docker Build', trust: 'TrustPermission'],
+                  [repo: 'docker-mirrorbits', owner: 'jenkins-infra', name: 'Mirrorbits Docker Build', trust: 'TrustPermission'],
+                  [repo: 'docker-datadog', owner: 'jenkins-infra', name: 'Datadog Docker Build', trust: 'TrustPermission'],
+                  [repo: 'docker-terraform', owner: 'jenkins-infra', name: 'Terraform Docker Build', trust: 'TrustPermission'],
+                  [repo: 'docker-builder', owner: 'jenkins-infra', name: "Docker Builder's Docker Build", trust: 'TrustPermission'],
+                  [repo: 'aws', owner: 'jenkins-infra', name: 'AWS Infra'],
```
Comment (timja) [inline config/default/jenkins-infra.yaml:159]: missing trust strategy

File: config/default/jenkins-infra.yaml [hunk 3]
```diff
@@ -149,104 +150,48 @@ jenkins:
...
+                      def traits = node / 'sources' / 'data' / 'jenkins.branch.BranchSource' / 'source' / 'traits'
+                      traits << 'org.jenkinsci.plugins.github__branch__source.BranchDiscoveryTrait' {
+                        strategyId(1)
+                      }
+                      traits << 'org.jenkinsci.plugins.github__branch__source.OriginPullRequestDiscoveryTrait' {
+                        strategyId(1)
+                      }
+                      traits << 'org.jenkinsci.plugins.github__branch__source.ForkPullRequestDiscoveryTrait' {
+                        strategyId(1)
+                        trust(class: 'org.jenkinsci.plugins.github_branch_source.ForkPullRequestDiscoveryTrait$' + config.trust)
```
Comment (timja) [inline config/default/jenkins-infra.yaml:186]: does this actually need to be configurable atm

File: config/default/jenkins-infra.yaml [hunk 4]
```diff
@@ -149,104 +150,48 @@ jenkins:
...
-                    workflowBranchProjectFactory {
-                      scriptPath('Jenkinsfile_k8s')
+                [
+                  [repo: 'docker-packer', owner: 'jenkins-infra', name: 'Packer Docker Build', trust: 'TrustPermission'],
+                  [repo: 'docker-mirrorbits', owner: 'jenkins-infra', name: 'Mirrorbits Docker Build', trust: 'TrustPermission'],
+                  [repo: 'docker-datadog', owner: 'jenkins-infra', name: 'Datadog Docker Build', trust: 'TrustPermission'],
+                  [repo: 'docker-terraform', owner: 'jenkins-infra', name: 'Terraform Docker Build', trust: 'TrustPermission'],
+                  [repo: 'docker-builder', owner: 'jenkins-infra', name: "Docker Builder's Docker Build", trust: 'TrustPermission'],
+                  [repo: 'aws', owner: 'jenkins-infra', name: 'AWS Infra'],
+                ].each{ config ->
```
Comment (timja) [inline config/default/jenkins-infra.yaml:160]: would it be better to just configure a github organisation job? so we don't have to update this every time a new repo comes along?

File: config/default/jenkins-infra.yaml [hunk 5]
```diff
@@ -217,36 +170,48 @@ jenkins:
...
-                  factory {
-                    workflowBranchProjectFactory {
-                      scriptPath('Jenkinsfile_k8s')
+                [
+                  [repo: 'docker-packer', owner: 'jenkins-infra', name: 'Packer Docker Build', trust: 'TrustPermission'],
+                  [repo: 'docker-mirrorbits', owner: 'jenkins-infra', name: 'Mirrorbits Docker Build', trust: 'TrustPermission'],
+                  [repo: 'docker-datadog', owner: 'jenkins-infra', name: 'Datadog Docker Build', trust: 'TrustPermission'],
+                  [repo: 'docker-terraform', owner: 'jenkins-infra', name: 'Terraform Docker Build', trust: 'TrustPermission'],
+                  [repo: 'docker-builder', owner: 'jenkins-infra', name: "Docker Builder's Docker Build", trust: 'TrustPermission'],
+                  [repo: 'aws', owner: 'jenkins-infra', name: 'AWS Infra', trust: 'TrustNobody'],
```
Comment (timja) [inline config/default/jenkins-infra.yaml:179]: could you explain why this is trust nobody please?

Comment (garethjevans) [reply config/default/jenkins-infra.yaml:179]: this repo is very much experimental atm, we'd like to change the permission level once we've worked out what we can safely run on a PR

Comment (timja) [reply config/default/jenkins-infra.yaml:179]: permission level doesn't have anything to do with what you can run though?

permission level dictates who is allowed to change a Jenkinsfile and have it validated.

everywhere else we use 'write access to the repo' as the who is allowed to change the file.

sure someone with write access _could_ change the file but i'm not sure this is really the right approach...

development also seems to be the wrong phase to add something like this...

Comment (garethjevans) [reply config/default/jenkins-infra.yaml:179]: So keeping `TrustPermission` for all repos would be sufficient?

Comment (timja) [reply config/default/jenkins-infra.yaml:179]: yes, then I would question why it's configurable at all, it's not something that is normally changed, the default is fine in 99% of cases.

**Pull request closed without merging**